### PR TITLE
feat: Add remove liquidity functions to bLUSD Zap contract

### PR DESCRIPTION
**Pull request by bingen**

### CODE REVIEW

File: LUSDChickenBonds/src/BLUSDLPZap.sol [hunk 1]
```diff
@@ -96,4 +96,54 @@ contract BLUSDLPZap {
...
+        // pull LP tokens
+        bLUSDLUSD3CRVLPToken.transferFrom(msg.sender, address(this), _lpAmount);
+
+        // All those balances should be zero, but just in case:
+        uint256 initialLUSD3CRVBalance = lusd3CRVPool.balanceOf(address(this));
+        uint256 initialBLUSDBalance = bLUSDToken.balanceOf(address(this));
+
+        // withdraw bLUSD/LUSD-3CRV
+        bLUSDLUSD3CRVLPToken.approve(address(bLUSDLUSD3CRVPool), _lpAmount);
+        bLUSDLUSD3CRVPool.remove_liquidity(_lpAmount, [_minBLUSD, 0], false, address(this));
```
Comment (RickGriff) [inline LUSDChickenBonds/src/BLUSDLPZap.sol:110]: Do we also need a minimum for the received LUSDCRV-f token quantity here? I guess this pool could be more easily manipulated than larger Curve pools (especially in early days), but haven't thought about attacks / extractable profit for it

File: LUSDChickenBonds/src/BLUSDLPZap.sol [hunk 2]
```diff
@@ -96,4 +96,54 @@ contract BLUSDLPZap {
...
+        uint256 bLUSDAmount = bLUSDToken.balanceOf(address(this)) - initialBLUSDBalance;
+        uint256 lusd3CRVAmount = lusd3CRVPool.balanceOf(address(this)) - initialLUSD3CRVBalance;
+
+        // withdraw LUSD from LUSD/3pool, and tranfer it to sender
+        if (lusd3CRVAmount > 0) {
+            lusd3CRVPool.remove_liquidity_one_coin(
+                lusd3CRVAmount,
+                0,
+                _minLUSD,
```
Comment (RickGriff) [inline LUSDChickenBonds/src/BLUSDLPZap.sol:119]: ...although, I guess this `_minLUSD` implicitly gives us a minimum LUSD3CRV quantity for the above withdrawal. 

Unless the bLUSD-LUSD3CRV pool is so imbalanced that we withdraw 0 LUSD3CRV in L110 (in which case L119 won't get reached), which seems unlikely/impossible.  So almost certainly not an issue.

**Review by RickGriff — APPROVED** [review]
Looks good!

File: LUSDChickenBonds/src/BLUSDLPZap.sol (hunk 2 above)
Comment (bingen) [reply LUSDChickenBonds/src/BLUSDLPZap.sol:119]: Yes, that was the idea, use the `_minLUSD` to implicitly enforce the other one.
But good point about getting zero LUSD3CRV. I guess it doesn’t hurt to add a check for that.

**Pull request merged**